### PR TITLE
[corlib] Make by-ref struct actually ref now C# supports them.

### PR DESCRIPTION
**Pull request by marek-safar**
This is technically breaking change but the types are already specialized
in C# compiler to be ref structs therefore the breaking change is more
theoretical than practical

Part of #11430

### CODE REVIEW

**Comment by marek-safar** [comment]
@monojenkins apidiff

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff

**Comment by lambdageek** [comment]
Nice.

I hardcoded `RuntimeArgumentHandle` and `TypedReference` as `IsByRefLike` in the runtime.  Can we drop that now, or will that break things?

**Comment by marek-safar** [comment]
@lambdageek yes, that can be removed now

File: mcs/class/corlib/System/RuntimeArgumentHandle.cs [hunk 1]
```diff
@@ -31,13 +31,9 @@
...
 //
 
-using System;
-using System.Runtime.InteropServices;
-
 namespace System
 {
-	[ComVisible (true)]
-	public struct RuntimeArgumentHandle
+	public ref struct RuntimeArgumentHandle
```
Comment (benaadams) [inline mcs/class/corlib/System/RuntimeArgumentHandle.cs]: Could also be `readonly`?
```csharp
public readonly ref struct RuntimeArgumentHandle
{
    internal readonly IntPtr args;
}
```

**Pull request merged**